driver/SHiELD/gfdl_cloud_microphys.F90

```diff
@@ -0,0 +1,4699 @@
+                u_dt (i, j, k) = u_dt (i, j, k) + (u1 (k) - u0 (k)) * rdt
+                v_dt (i, j, k) = v_dt (i, j, k) + (v1 (k) - v0 (k)) * rdt
+            enddo
+        endif
+
+        if (do_sedi_w) then
+            do k = ktop, kbot
+                w (i, j, k) = w1 (k)
+            enddo
+        endif
+
+        ! -----------------------------------------------------------------------
+        ! update moist air mass (actually hydrostatic pressure)
+        ! convert to dry mixing ratios
+        ! -----------------------------------------------------------------------
+
+        do k = ktop, kbot
+            omq = dp1 (k) / dp0 (k)
+            qv_dt (i, j, k) = qv_dt (i, j, k) + rdt * (qvz (k) - qv0 (k)) * omq
+            ql_dt (i, j, k) = ql_dt (i, j, k) + rdt * (qlz (k) - ql0 (k)) * omq
+            qr_dt (i, j, k) = qr_dt (i, j, k) + rdt * (qrz (k) - qr0 (k)) * omq
+            qi_dt (i, j, k) = qi_dt (i, j, k) + rdt * (qiz (k) - qi0 (k)) * omq
+            qs_dt (i, j, k) = qs_dt (i, j, k) + rdt * (qsz (k) - qs0 (k)) * omq
+            qg_dt (i, j, k) = qg_dt (i, j, k) + rdt * (qgz (k) - qg0 (k)) * omq
+            cvm = c_air + qvz (k) * c_vap + (qrz (k) + qlz (k)) * c_liq + (qiz (k) + qsz (k) + qgz (k)) * c_ice
+            pt_dt (i, j, k) = pt_dt (i, j, k) + rdt * (tz (k) - t0 (k)) * cvm / cp_air
+        enddo
+
+        ! -----------------------------------------------------------------------
+        ! update cloud fraction tendency
+        ! -----------------------------------------------------------------------
+
+        do k = ktop, kbot
+            if (do_qa) then
+                qa_dt (i, j, k) = 0.
+            else
+                qa_dt (i, j, k) = qa_dt (i, j, k) + rdt * (qaz (k) / real (ntimes) - qa0 (k))
+            endif
+        enddo
+
+        ! -----------------------------------------------------------------------
+        ! fms diagnostics:
+        ! -----------------------------------------------------------------------
+
+        ! if (id_cond > 0) then
+        ! do k = ktop, kbot ! total condensate
+        ! cond (i) = cond (i) + dp1 (k) * (qlz (k) + qrz (k) + qsz (k) + qiz (k) + qgz (k))
+        ! enddo
+        ! endif
+        !
+        ! if (id_vtr > 0) then
+        ! do k = ktop, kbot
+        ! vt_r (i, j, k) = vtrz (k)
+        ! enddo
+        ! endif
+        !
+        ! if (id_vts > 0) then
+        ! do k = ktop, kbot
+        ! vt_s (i, j, k) = vtsz (k)
+        ! enddo
+        ! endif
+        !
+        ! if (id_vtg > 0) then
+        ! do k = ktop, kbot
+        ! vt_g (i, j, k) = vtgz (k)
+        ! enddo
+        ! endif
+        !
+        ! if (id_vts > 0) then
+        ! do k = ktop, kbot
+        ! vt_i (i, j, k) = vtiz (k)
+        ! enddo
+        ! endif
+        !
+        ! if (id_droplets > 0) then
+        ! do k = ktop, kbot
+        ! qn2 (i, j, k) = ccn (k)
+        ! enddo
+        ! endif
```

Comment:
When the SHiELD system was created, it used the IPD interface to the atmospheric physical parameterizations.  This IPD imposed a different method for handling diagnostics and the direct use of send_data became deprecated.  The GFDL and FMS specific methods for managing diagnostics are still in the codebase to ensure inter-operability with GFDL models.  This could be better handled via a macro, but for now, the codeowner chose to use comments.